sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageConverter.cs

```diff
@@ -139,13 +142,20 @@ private static ArraySegment<byte> ReadStreamToArraySegment(Stream stream)
                 return new ArraySegment<byte>();
             }
 
-            using var memStream = new MemoryStream(StreamBufferSizeInBytes);
-            stream.CopyTo(memStream, StreamBufferSizeInBytes);
-
-            return new ArraySegment<byte>(memStream.ToArray());
+            switch (stream)
+            {
+                case BufferListStream bufferListStream:
+                    return bufferListStream.ReadBytes((int)stream.Length);
+                default:
```

Comment:
I did not have enough time to check whether we could even eliminate this branch. If we can always assume `BufferListStream` we can even squeeze more out of this due to eliminated branching. So if anyone of the maintainers can answer me the question and save me a bunch of investigation I would love that :D

Reply:
It looks like it currently is a BufferListStream, but the AmqpMessage.ToStream() method returns a `Stream`, so I don't think we can take a dependency on that.